pkg/types/var.go

```diff
@@ -71,6 +72,19 @@ func (v *Var) defaulting() {
 	}
 }
 
+// VarEquals returns true if var a and b are Equals.
+func (v Var) DeepEqual(other Var) bool {
+	v.ObjRef.GVK()
+	if v.FieldRef.FieldPath == "" {
+		v.FieldRef.FieldPath = "metadata.name"
+	}
+	other.ObjRef.GVK()
```

Comment:
How about calling `GVK()` inside `defaulting`? Then you don't need to write this deepequal function

Reply:
@Liujingfang1 Let me try that first. Issue was there was no public method to run the defaulting. Come back to you.

Reply:
@Liujingfang1 This current version of the PR is using var1.Defaulting(), var2.Defaulting() in order to be able to run reflect.DeepEqual(var1,var2). Not sure it is better.

Reply:
@Liujingfang1 Second commit is showing that we have to invoke Defaulting() from outside the var struct. Third commit is using DeepEqual instead. Looks better to my opinion